src/txn/txn_recover.c

```diff
@@ -522,6 +532,15 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 done:	FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
 err:	WT_TRET(__recovery_free(&r));
 	__wt_free(session, config);
+
+	/*
+	 * Destroy the eviction threads that were started in support of
+	 * recovery.  They will be restarted once the lookaside table is
+	 * created.
+	 */
+	if (eviction_started)
+		WT_TRET(__wt_evict_destroy(session));
+
```

Comment:
@keithbostic It was me that pushed @michaelcahill to encapsulate recovery in a single function. Since the eviction threads are being started and stopped solely for recovery I thought it made more sense to do that within the `__wt_txn_recover` method.

Why would you prefer the calls to be outside the function? Are you thinking something like `WT_WITH_EVICTION_THREADS(__wt_txn_recover(...));`  (psuedo code, I doubt we'd actually use a macro) is clearer?